src/commands/qrcode.cpp

```diff
@@ -23,25 +23,172 @@
 #include <iostream>
 #include <bitcoin/explorer/define.hpp>
 
+#if defined WITH_QRENCODE && defined WITH_LIBPNG
+#include "png.h"
+#include "qrencode.h"
+#endif
+
 using namespace bc::explorer;
 using namespace bc::explorer::commands;
 
+/*
+  The following constants and write_png method are based on:
+
+  https://github.com/fukuchi/libqrencode/blob/3.4/qrenc.c
+*/
+
+#if defined WITH_QRENCODE && defined WITH_LIBPNG
+static constexpr int dpi = 72;
+static constexpr int version = 0;
+static constexpr int margin = 2;
+static constexpr QRencodeMode hint = QR_MODE_8;
+static constexpr QRecLevel level = QR_ECLEVEL_L;
+static constexpr int casesensitive = 1;
+static constexpr int INCHES_PER_METER = (100.0 / 2.54);
+static constexpr unsigned int fg_color[4] = {0, 0, 0, 255};
+static constexpr unsigned int bg_color[4] = {255, 255, 255, 255};
```

Comment:
The above options should eventually be exposed as command line options. This would require the definition of primitives for enumerated type serialization (QRencodeMode & QRecLevel) and for color definition.

Prefer use of `int32_t` and `uint32_t` as a matter of style (prefer explicit).